collab-database/src/database.rs

```diff
@@ -425,15 +421,37 @@ impl Database {
       .collect()
   }
 
-  pub fn create_field(&self, field: Field) {
+  /// Creates a new field, appends the field_id to the end of field order
+  /// and adds a field setting
+  pub fn create_field(
+    &self,
+    field: Field,
+    field_settings_by_layout: HashMap<DatabaseLayout, FieldSettings>,
+  ) {
     self.root.with_transact_mut(|txn| {
-      self.create_field_with_txn(txn, field);
+      self.create_field_with_txn(txn, field, &field_settings_by_layout);
     })
   }
 
-  pub fn create_field_with_txn(&self, txn: &mut TransactionMut, field: Field) {
+  pub fn create_field_with_txn(
+    &self,
+    txn: &mut TransactionMut,
+    field: Field,
+    field_settings_by_layout: &HashMap<DatabaseLayout, FieldSettings>,
+  ) {
     self.views.update_all_views_with_txn(txn, |update| {
-      update.push_field_order(&field);
+      let field_id = field.id.clone();
+      update
+        .push_field_order(&field)
+        .update_field_settings_for_fields(
+          vec![field_id],
+          |field_id, field_setting_update, layout_ty| {
+            field_setting_update.update(
+              field_id,
+              field_settings_by_layout.get(&layout_ty).unwrap().clone(),
```

Comment:
Is it need to get the single setting of the field by its id? Because the field_settings_by_layout.get(&layout_ty) return all the fields' settings by database layout type.